code/code/misc/riding.cc

```diff
@@ -823,8 +841,24 @@ int MountEgoCheck(TBeing *ch, TBeing *horse)
     check = horse->GetMaxLevel();
     check += dynamic_cast<TMonster *>(horse)->anger() / 30;
     check -= ch->GetMaxLevel();
+
     if (ch->doesKnowSkill(SKILL_RIDE))
       check -= ch->getSkillValue(SKILL_RIDE)/10;
+
+    // Bonus for proficiency in advanced riding disc.
+    if (ch->doesKnowSkill(SKILL_ADVANCED_RIDING))
+      check -= ch->getSkillValue(SKILL_ADVANCED_RIDING)/8;
+
+    // Bonus for proficiency in type-specific mount ability
+    if (ch->doesKnowSkill(SKILL_RIDE_WINGED) && horse->mountSkillType() == SKILL_RIDE_WINGED)
+      check -= ch->getSkillValue(SKILL_RIDE_WINGED)/6;
+    else if (ch->doesKnowSkill(SKILL_RIDE_DOMESTIC) && horse->mountSkillType() == SKILL_RIDE_DOMESTIC)
+      check -= ch->getSkillValue(SKILL_RIDE_DOMESTIC)/6;
+    else if (ch->doesKnowSkill(SKILL_RIDE_NONDOMESTIC) && horse->mountSkillType() == SKILL_RIDE_NONDOMESTIC)
+      check -= ch->getSkillValue(SKILL_RIDE_NONDOMESTIC)/6;
+    else if (ch->doesKnowSkill(SKILL_RIDE_EXOTIC) && horse->mountSkillType() == SKILL_RIDE_EXOTIC)
+      check -= ch->getSkillValue(SKILL_RIDE_EXOTIC)/6;
```

Comment:
This code is fine, but it's not how I'd write it. I'd write something like this:

```cpp
for (const auto skill : {SKILL_RIDE_WINGED, SKILL_RIDE_DOMESTIC, SKILL_RIDE_NONDOMESTIC, SKILL_RIDE_EXOTIC})
  check -= ch->doesKnowSkill(SKILL_RIDE_WINGED) ? ch->getSkillValue(SKILL_RIDE_WINGED)/6 : 0;
```

Or else I'd see if there's a getSkillValue variant that returns 0 if you don't know the skill, and get rid of the ternary.

Reply:
Pretty nifty.  I haven't used this before (and it'll probably be a while before I think to use it again just based on comfort level), but I reworked the code just to get more familiar with it.  Tested and it seems to function the same way as before (properly).  

I did re-order some of checks just because they make a bit more sense to me this way in the game, namely, we perform the check on if you can ride a flying mount first (because this can be a hard fail) and then we later attempt the ride check for being in combat/tanking.  In the original order, you would get a message about failing to ride because it's hard while tanking... and then if you passed the check, it would just say you can't ride a flying mount.  In short, I moved the hard check before the soft check.